### PR TITLE
sm3: add OSCCA SM3 secure hash

### PR DESCRIPTION
**Pull request by uudiin**
Add OSCCA SM3 secure hash (OSCCA GM/T 0004-2012 SM3) generic
hash transformation.

### CODE REVIEW

File: sm3/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+[package]
+name = "sm3"
```
Comment (tarcieri) [inline sm3/Cargo.toml:2]: It looks like there is already a published `sm3` crate:

https://crates.io/crates/sm3

The repo/source code is here:

https://github.com/andelf/rust-sca/tree/master/sm3

Perhaps we could ask the author about upstreaming it.

Comment (uudiin) [reply sm3/Cargo.toml:2]: Putting these crates on RustCrypto would be a better choice, I will try to contact the original author. Thanks.

Comment (andelf) [reply sm3/Cargo.toml:2]: I've invited @newpavlov as an owner of `sm3` on crates.io.

Feel free to overwrite it with RustCrypto crates. 😁

Also, I've invited @uudiin to `rust-sca` repo. You can update any READMEs and links.

Comment (tarcieri) [reply sm3/Cargo.toml:2]: Wow, awesome, thank you!

**Comment by tarcieri** [comment]
@newpavlov can you add the `RustCrypto/hashes` team as an owner of the `sm3` crate?

**Comment by newpavlov** [comment]
@tarcieri 
Done!

**Pull request merged**